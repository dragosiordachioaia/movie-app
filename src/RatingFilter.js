import React, { Component } from "react";

import "./RatingFilter.scss";

import RangeSlider from "./RangeSlider";

export default class RatingFilter extends Component {
  render() {
    return (
      <div className="rating-filter">
        <p className="label" />
        <RangeSlider
          min={0}
          max={10}
          step={0.5}
          value={this.props.minRating}
          onChange={this.props.onChange}
        />
      </div>
    );
  }
}
