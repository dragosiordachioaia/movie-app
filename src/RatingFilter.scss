@import "variables";

.rating-filter {
  display: flex;
  margin-top: 1em;
  display: block;
  margin-bottom: 2.5em;

  .label {
    width: 100%;
    display: block;
    margin-right: 0em;

    &::after {
      content: "Minimum rating";
    }
  }
}
