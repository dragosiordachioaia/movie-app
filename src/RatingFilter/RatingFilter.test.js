import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import RatingFilter from "./RatingFilter";

jest.mock("RangeSlider/RangeSlider", () => "RangeSlider");

let component;
let instance;

beforeEach(() => {
  component = renderer.create(<RatingFilter />);
  instance = component.getInstance();
});

it("renders correctly", () => {
  instance.props = { expanded: false };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
