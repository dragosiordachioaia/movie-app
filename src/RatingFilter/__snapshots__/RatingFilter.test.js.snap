// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rating-filter"
>
  <p
    className="label"
  >
    Minimum rating
  </p>
  <RangeSlider
    max={10}
    min={0}
    step={0.5}
  />
</div>
`;
